### DISCUSSION TITLE
bw2calc.errors.NonsquareTechnosphere

### DISCUSSION THREAD

**Discussion by danielembc** [LCA-ActivityBrowser/activity-browser#1227] [Q&A]
Hi everyone,

After modifying a few processes, I receive the following error: bw2calc.errors.NonsquareTechnosphere: Technosphere matrix is not square: 21329 activities (columns) and 21330 products (rows). Use LeastSquaresLCA to solve this system, or fix the input data

I followed the steps here https://stackoverflow.com/questions/43498326/find-a-product-that-causes-a-non-square-technosphere-matrix-with-brightway2 for a solution, but I can't see what is the process/activity generating the issue.

Can someone help me with this?

Thank you!

**Comment by marc-vdm** [comment]
Hi DC,

The SO solution is for 1 database, did you check all databases in your calculation setup? And how did you create these processes? It should not be possible to create processes in AB with =/= 1 products, but perhaps you found a bug 😅

**Comment by danielembc** [comment]
Hi Marc,

Thanks for your reply! I did all the steps you mentioned.

The problem started after copying and pasting a process, deleting the
original and renaming it as the original one. What solved it: deleting all
calculation setups and redoing them (#sad 😅).

I think it is indeed a bug! I hope this solution helps others in the
future.

Kind regards,
Daniele

**Reply by marc-vdm** [reply]
Happy it''s not a problem for you anymore! We'll try an repeat your steps to see if we can replicate the problem so we can fix it. So thanks a lot for sharing!
